bundle/outputs.go

```diff
@@ -6,3 +6,17 @@ type Output struct {
 	Description string   `json:"description,omitempty" yaml:"description,omitempty"`
 	Path        string   `json:"path" yaml:"path"`
 }
+
+// AppliesTo returns a boolean value specifying whether or not
+// the Output applies to the provided action
+func (output *Output) AppliesTo(action string) bool {
+	if len(output.ApplyTo) == 0 {
```

Comment:
It feels like there's a pattern with the Parameter type, I think we should factorize the code, maybe using embedded type?

Reply:
It's true... I could use some pointers/examples here.  I've tried a couple different ways (couldn't figure out how to prevent embedded struct from printing if empty on Marshal; couldn't quite figure out how to use an interface to avoid code duplication) but had to admit defeat.  Help would be welcome!  Could be in a follow-up as well, if that'd work.

Reply:
Ok let's keep it like this for now 👍